nomad/task_group_host_volume_claim_endpoint.go

```diff
@@ -0,0 +1,179 @@
+)
+
+// TaskGroupHostVolumeClaim is the server RPC endpoint for task group volume claims
+type TaskGroupHostVolumeClaim struct {
+	srv    *Server
+	ctx    *RPCContext
+	logger hclog.Logger
+}
+
+func NewTaskGroupVolumeClaimEndpoint(srv *Server, ctx *RPCContext) *TaskGroupHostVolumeClaim {
+	return &TaskGroupHostVolumeClaim{srv: srv, ctx: ctx, logger: srv.logger.Named("task_group_host_volume_claim")}
+}
+
+func (tgvc *TaskGroupHostVolumeClaim) List(args *structs.TaskGroupVolumeClaimListRequest, reply *structs.TaskGroupVolumeClaimListResponse) error {
+	authErr := tgvc.srv.Authenticate(tgvc.ctx, args)
+	if done, err := tgvc.srv.forward(structs.TaskGroupHostVolumeClaimListRPCMethod, args, args, reply); done {
+		return err
+	}
+	tgvc.srv.MeasureRPCRate("task_group_volume_claim", structs.RateMetricList, args)
+	if authErr != nil {
+		return structs.ErrPermissionDenied
+	}
+	defer metrics.MeasureSince([]string{"nomad", "task_group_volume_claim", "list"}, time.Now())
+
+	allowClaim := acl.NamespaceValidator(acl.NamespaceCapabilityHostVolumeRead)
+	aclObj, err := tgvc.srv.ResolveACL(args)
+	if err != nil {
+		return err
+	}
+	if !allowClaim(aclObj, args.RequestNamespace()) {
+		return structs.ErrPermissionDenied
+	}
+
+	ns := args.RequestNamespace()
+
+	opts := blockingOptions{
+		queryOpts: &args.QueryOptions,
+		queryMeta: &reply.QueryMeta,
+		run: func(ws memdb.WatchSet, stateStore *state.StateStore) error {
+			var iter memdb.ResultIterator
+			var err error
+
+			switch {
+			case args.JobID != "":
+				iter, err = stateStore.TaskGroupHostVolumeClaimsByJobID(ws, args.JobID)
+			case args.TaskGroup != "":
+				iter, err = stateStore.TaskGroupHostVolumeClaimsByTaskGroup(ws, args.TaskGroup)
+			case args.VolumeName != "":
+				iter, err = stateStore.TaskGroupHostVolumeClaimsByVolumeName(ws, args.VolumeName)
+			default:
+				iter, err = stateStore.GetTaskGroupHostVolumeClaims(ws)
+			}
+			if err != nil {
+				return err
+			}
+
+			tokenizer := paginator.NewStructsTokenizer(iter,
+				paginator.StructsTokenizerOptions{
+					WithNamespace: true,
+					WithID:        true,
+				},
+			)
+
+			filters := []paginator.Filter{
+				paginator.GenericFilter{
+					Allow: func(raw any) (bool, error) {
+						claim := raw.(*structs.TaskGroupHostVolumeClaim)
+						// empty prefix doesn't filter
+						if !strings.HasPrefix(claim.ID, args.Prefix) {
+							return false, nil
+						}
+						if args.JobID != "" && claim.JobID != args.JobID {
+							return false, nil
+						}
+						if args.TaskGroup != "" && claim.TaskGroupName != args.TaskGroup {
+							return false, nil
+						}
+						if args.VolumeName != "" && claim.VolumeName != args.VolumeName {
+							return false, nil
+						}
+
+						if ns != structs.AllNamespacesSentinel &&
+							claim.Namespace != ns {
+							return false, nil
+						}
+
+						allowClaim := acl.NamespaceValidator(acl.NamespaceCapabilityHostVolumeRead)
+						return allowClaim(aclObj, ns), nil
```

Comment:
I think this needs to be the object's namespace, right?

```suggestion
						return allowClaim(aclObj, claim.Namespace), nil
```